test/contract/db.go

```diff
@@ -0,0 +1,43 @@
+package contract
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+func WithDatabase(prefix string) OptResourceFunc {
+	return func(ctx context.Context, resources *TestResources) (*TestResources, error) {
+		//
+		auth := options.Credential{
+			AuthMechanism: "SCRAM-SHA-1",
```

Comment:
Why using SHA-1?

Reply:
SCRAM-SHA-256 never works in my tests. SCRAM-SHA-1 works sometimes.

Reply:
Still SCRAM-SHA-256 never works, but we know SCRAM-SHA-1 was failing when tested to early, as the changes to the cluster user took some time to be applied after the call returned with success.